app/src/main/java/chat/rocket/android/chatroom/adapter/ActionsListAdapter.kt

```diff
@@ -0,0 +1,73 @@
+package chat.rocket.android.chatroom.adapter
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.RecyclerView
+import chat.rocket.android.R
+import chat.rocket.android.util.extensions.inflate
+import chat.rocket.core.model.attachment.actions.Action
+import chat.rocket.core.model.attachment.actions.ButtonAction
+import com.facebook.drawee.backends.pipeline.Fresco
+import kotlinx.android.synthetic.main.item_action_button.view.*
+import timber.log.Timber
+
+class ActionsListAdapter(actions: List<Action>, var actionAttachmentOnClickListener: ActionAttachmentOnClickListener) : RecyclerView.Adapter<ActionsListAdapter.ViewHolder>() {
+
+    var actions: List<Action> = actions
+
+    inner class ViewHolder(var layout: View) : RecyclerView.ViewHolder(layout) {
+        lateinit var action: ButtonAction
+
+        private val onClickListener = View.OnClickListener {
+            actionAttachmentOnClickListener.onActionClicked(it, action)
+        }
+
+        init {
+            with(itemView) {
```

Comment:
This is a good usage of `with` with more than one block code being used. Less than that is useless.

👍